tree-sitter-stack-graphs/src/lib.rs

```diff
@@ -436,16 +433,17 @@ impl StackGraphLanguage {
     /// (The source file must be implemented in this language, otherwise you'll probably get a
     /// parse error.)
     pub fn build_stack_graph_into(
-        &mut self,
+        &self,
         stack_graph: &mut StackGraph,
         file: Handle<File>,
         source: &str,
         globals: &mut Variables,
+        cancellation_flag: &dyn CancellationFlag,
     ) -> Result<(), LoadError> {
-        let tree = self
-            .parser
-            .parse(source, None)
-            .ok_or(LoadError::ParseError)?;
+        let mut parser = Parser::new();
+        parser.set_language(self.language)?;
+        unsafe { parser.set_cancellation_flag(cancellation_flag.flag()) };
```

Comment:
fffuuuuuuuuuuuuu

Now I see the issue. This is the most expedient approach I can think of. The clean eventual solution would be to pull this trait out into a separate crate that `tree-sitter`, `stack-graphs`, `tree-sitter-graph`, and `tree-sitter-stack-graph` can all depend on. But that's too involved to worry about right now.

Can you add a `TODO` comment to that effect?  Other than that I think this is good to go